src/index.ts

```diff
@@ -4,38 +4,58 @@ import * as core from "@actions/core";
 import * as exec from "@actions/exec";
 import { matlab } from "run-matlab-command-action";
 import * as buildtool from "./buildtool";
+import * as buildSummary from "./buildSummary";
 
 /**
  * Gather action inputs and then run action.
  */
 async function run() {
-    const platform = process.platform;
-    const architecture = process.arch;
-    const workspaceDir = process.cwd();
-
-    const options: buildtool.RunBuildOptions = {
-        Tasks: core.getInput("tasks"),
-        BuildOptions: core.getInput("build-options"),
-    };
-
-    const command = buildtool.generateCommand(options);
-    const startupOptions = core.getInput("startup-options").split(" ");
-
-    const helperScript = await matlab.generateScript(workspaceDir, command);
-    const execOptions  = { env: {
-        ...process.env,
-        "MW_MATLAB_BUILDTOOL_DEFAULT_PLUGINS_FCN_OVERRIDE":"ciplugins.github.getDefaultPlugins",
-    }};
-
-    await matlab.runCommand(
-        helperScript,
-        platform,
-        architecture,
-        (cmd,args)=>exec.exec(cmd,args,execOptions),
-        startupOptions
-    );
+    try {
```

Comment:
I think that we should try to avoid wrapping the entire function in a `try` block if possible. I preferred the version that you included in your reply to @mcafaro, did there end up being something wrong with that?

Reply:
How would we handdle the exception occurred for rest of the code in run() function where we need to set the build to be failed in case of any errors if we did not wrapped the entire code in try {}. since I am calling only `run()`

Reply:
First, I would remove the call to `buildSummary.processAndDisplayBuildSummary` on line 43. Then I was thinking something like this:

```typescript
run().catch(e => {
    ...
}).finally(() => {
    try {
        buildSummary..processAndDisplayBuildSummary();
    } catch (e) {
         ...
    }
});
```

I'm also not totally sure whether we should mark the build as failed if `run` succeeds but then uploading the summary fails. I'm trying to decide what I would expect / want as a user.

Reply:
Yeah but is't `run().catch is similar to run logic in try{} i thought its same here.

Reply:
i have implemented it in finally and set the failed only at catch. don't want to setFailed on summary display